File: java/headFirstDesignPatterns/chap1/duckExample/src/br/com/duck/ToyDuck.java
package br.com.duck;

import br.com.duck.fly.FlyNoWay;
import br.com.duck.quack.MuteQuack;

public class ToyDuck extends Duck{
    public ToyDuck(){
        flyBehavior = new FlyNoWay();
        quackBehavior = new MuteQuack();
    }

    @Override
    public void notFly() {}
}

File: c_cpp/headFirst_designPatterns/chap1/duckExample/src/ToyDuck.cpp
/*
 * ToyDuck.cpp
 *
 */

#include "ToyDuck.hpp"

namespace ducks{

ToyDuck::ToyDuck(){
    flyBehavior = new fly::FlyNoWay();
    quackBehavior = new quack::MuteQuack();
}

ToyDuck::~ToyDuck(){
    delete flyBehavior;
    delete quackBehavior;
}

void ToyDuck::notFly(){}

}
